apps/hannk/interpreter/allocation_planner.h

```diff
@@ -0,0 +1,53 @@
+#ifndef HANNK_MEMORY_PLANNER_H
+#define HANNK_MEMORY_PLANNER_H
+
+#include <vector>
+
+namespace hannk {
+
+// AllocationPlanner is used to plan a series of allocations in which we can
+// overlap blocks that don't have any lifespan in common. At present,
+// the implementation is incredibly simpleminded (it never overlaps anything)
+// but will be replaced by something smarter soon (likely a basic greedy allocator).
+class AllocationPlanner {
+public:
+    // All blocks allocated will be aligned to (at least) this amount.
+    explicit AllocationPlanner(size_t alignment);
+
+    // Specify a block's size and lifetime. The order this is called implicitly
+    // assigns an index to the result, so the block information that's passed into the nth call of
```

Comment:
My rationale here was:
- if it doesn't promise to return an incrementing index, the caller will theoretically need to keep track of the result in a map (etc) to ensure the results are sound
- if it *does* promise to return an incrementing index, the return value is redundant since the caller can trivially compute it

That said, I don't think it's a big deal either way, so sure, I can make this change